File: android/app/src/main/kotlin/com/sopro/sopro/logging/LogEvent.kt
package com.sopro.sopro.logging

// Modelo completo de um evento de log — espelho do LogEvent Flutter.
// Criado pelo Logger, consumido pelo JsonFormatter e sinks de saída.
data class LogEvent(
    // Versão do schema JSON. Nunca preencher manualmente — Logger injeta via LoggerConfiguration.
    val schemaVersion: Int,
    val level: LogLevel,
    val message: String,
    val context: LogContext,
    // Payload já sanitizado por LogSanitizer antes de chegar aqui.
    val payload: Map<String, Any?>?,
    val durationMs: Long?,
    val exception: Throwable?,
    // Pré-serializado para evitar recalcular em cada sink de saída.
    val stackTraceString: String?,
)

File: lib/infrastructure/logging/core/log_event.dart
import 'log_context.dart';
import 'log_level.dart';

// Modelo completo de um evento de log.
// Criado pelo Logger e consumido pelo JsonLogFormatter e pelas sinks de saída.
class LogEvent {
  const LogEvent({
    required this.level,
    required this.message,
    required this.context,
    this.payload,
    this.durationMs,
    this.exception,
    this.stackTrace,
  });

  // Severidade do evento.
  final LogLevel level;

  // Mensagem principal — normalmente o event_type (ex.: 'geofence_enter').
  final String message;

  // Contexto de ambiente e sessão no momento do evento.
  final LogContext context;

  // Dados adicionais estruturados associados ao evento.
  // Já sanitizados pelo LogSanitizer antes de chegar aqui.
  final Map<String, dynamic>? payload;

  // Duração da operação em milissegundos (para eventos com tempo mensurável).
  final int? durationMs;

  // Exceção capturada, se o evento representa um erro.
  final Object? exception;

  // Stack trace associado à exceção.
  final StackTrace? stackTrace;
}
